public/js/app/validation.js

```diff
@@ -0,0 +1,43 @@
+const api_data = "https://api.faforever.com/data/player"
+
+// If username is taken update glyphicon and error message to inform the user
+function renderUserTaken(){
+	// this line shows text in the error area to inform about taken username
+	$("input[name='username']").siblings("div.help-block").eq(1).text("username is already in use");
+	// this changes the glyphicon to an ❌ insted of ✔️ indicating username cant be choosen
+	$("input[name='username']").siblings("span.glyphicon").addClass("glyphicon-remove");
+	// this marks the formgroup in red
+	$("input[name='username']").parents("div.form-group").addClass("has-error has-danger");
+}
+
+//clean up username taken message and glyphicon indication for next input attempt
+$("input[name='username']").focus(
+	function(event) {
+		// remove any error relating text, user now tries a new input
+		$("input[name='username']").siblings("div.help-block").eq(1).text("");
+		// dont show glyphicon with ❌ untill checked
+		$("input[name='username']").siblings("span.glyphicon").removeClass("glyphicon-remove");
+		// remove red markings untill input is checked agian
+		$("input[name='username']").parents("div.form-group").removeClass("has-error has-danger");
+	}
+);
+
+// After user changed the username check if its taken
+$("input[name='username']").change(function(event){
```

Comment:
Username must be longer than 3 chars. Also it would be good to have a delay in there (the user has not entered anything for ? seconds). I don't want to have 20 api queries while typing a 20 character long name.